File: 1-two-sum/1-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dict = {};
        for i in range(len(nums)):
            snum = target - nums[i];
            if (snum in dict.keys()):
                sindex = nums.index(snum)
                if(i!= sindex):
                    return sorted([i,sindex])
            
            dict.update({nums[i]:i})

File: 1-two-sum/1-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> h = new HashMap<>();
        
        for(int i=0;i<nums.length;i++)
        {
            int snum = target- nums[i];
            if(h.containsKey(snum))
            {
               return new int[] {i, h.get(snum)};
                
            }
             h.put(nums[i], i);
           
        }
         return null;
    }
}
